src/content/get-started/ch2-add-commands.md

```diff
@@ -0,0 +1,397 @@
+    information. This is expected behavior for now.
+    :::
+
+### Task 2: Implement the search command
+
+Next, implement a basic `search` command that takes an article title as
+input. As you build this functionality, you'll work with `List` manipulation,
+null checks, and string interpolation.
+
+1.  **Integrate the `search` command into `main`:** First, modify your `main`
+    function in `cli/bin/cli.dart` to include an `else if` branch that handles
+    the `search` command. For now, we'll just print a placeholder message.
+
+    ```dart
+    void main(List<String> arguments) {
+      if (arguments.isNotEmpty && arguments.first == 'version') {
+        print('Dartpedia CLI version $version');
+      } else if (arguments.isNotEmpty && arguments.first == 'help') {
+        printUsage();
+      } else if (arguments.isNotEmpty && arguments.first == 'search') {
+        // Add this new block:
+        print('Search command recognized!');
+      } else {
+        printUsage();
+      }
+    }
+    ```
+
+1.  **Test the new command:** Run your application with the `search`
+    command:
+
+    ```bash
+    dart bin/cli.dart search
+    ```
+
+    You should see:
+
+    ```bash
+    Search command recognized!
+    ```
+
+1.  **Define the `runApp` function:** The `search` command will eventually run
+    the core logic of your application by calling a function called `runApp`.
+    For now, it will print the arguments passed into search. Place this new
+    function below `main`.
+
+    ```dart
+    // ... (your existing main function)
+
+    void runApp(List<String>? arguments) { // Add this new function and add ? to arguments type
+      print('runApp received arguments: $arguments');
+    }
+
+    // ... (your existing printUsage() function)
+    ```
+
+    * The `List<String>? arguments` type signature means that the `arguments`
+        list itself can be `null`. This is important for null safety in Dart.
+
+1.  **Call `runApp` from `main`:** Now, modify the `search` command block in
+    `main` to call `runApp` and pass it any arguments that come after the
+    `search` command itself. We use `arguments.sublist(1)` to get all
+    arguments starting from the second one. If no arguments are provided after
+    `search`, we'll pass `null` to `runApp`.
+
+    ```dart
+    void main(List<String> arguments) {
+      if (arguments.isNotEmpty && arguments.first == 'version') {
+        print('Dartpedia CLI version $version');
+      } else if (arguments.isNotEmpty && arguments.first == 'help') {
+        printUsage();
+      } else if (arguments.isNotEmpty && arguments.first == 'search') { // Changed to 'search'
+        // Modify this block:
+        final inputArgs = arguments.length > 1 ? arguments.sublist(1) : null;
+        runApp(inputArgs);
+      } else {
+        printUsage();
+      }
+    }
+    ```
+
+    * **List manipulation:** `arguments.sublist(1)` creates a new list
```

Comment:
I think you should preface these bulleted lists with something like this:

Highlights from the preceding code: